Admin/pages/AdminApproval.php

```diff
@@ -35,13 +35,15 @@ protected function initInternal()
 
 		$this->pending_ids = $this->getPendingIds();
 
-		if (count($this->pending_ids) == 0)
+		if (count($this->pending_ids) === 0) {
 			$this->relocate();
+		}
 
 		$this->id = $this->app->initVar('id');
 
-		if ($this->id === null)
+		if ($this->id === null) {
 			$this->id = $this->getNextId();
+		}
```

Comment:
I see now, I had the logic of `$found = ($this->data_object === null);` mixed up.